File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/composite/player/FootballPlayer.java
// FootballPlayer.java

package com.bigboxcode.designpattern.composite.player;

public class FootballPlayer implements Player {
    private String name;
    private int age;

    private int goal;

    public FootballPlayer(String name, int age, int goal) {
        this.name = name;
        this.age = age;
        this.goal = goal;
    }

    @Override
    public void printDetails() {
        System.out.println("Game: Football");
        System.out.println("Name: " + name);
        System.out.println("Age: " + age);
        System.out.println("Goals: " + goal);
    }
}

File: design-patterns/ts/composite/player-group/football-player.ts
// football-player.ts

import Player from "./player";

class FootballPlayer implements Player {
    private name: string;
    private age: number;
    private goal: number;
    constructor(name: string, age: number, goal: number) {
        this.name = name;
        this.age = age;
        this.goal = goal;
    }

    printDetails() {
        console.log("Game: Football");
        console.log("Name: " + this.name);
        console.log("Age: " + this.age);
        console.log("Goals: " + this.goal);
    }
}

export default FootballPlayer;